keps/sig-node/2625-cpumanager-policies-thread-placement/README.md

```diff
@@ -309,64 +273,146 @@ We expect no impact. The new policies are opt-in and separated by the existing o
+
+###### Are there any tests for feature enablement/disablement?
+
+A specific e2e test will demonstrate that the default behaviour is preserved when the `CPUManagerPolicyOptions` feature gate is disabled, or when the feature is not used (2 separate tests)
 
 ### Rollout, Upgrade and Rollback Planning
 
-* **How can a rollout fail? Can it impact already running workloads?** Kubelet may fail to start. The kubelet may crash.
-* **What specific metrics should inform a rollback?**
-  - the number of pod ending up in Failed for SMTAlignmentError could be used to decide a rollback.
-* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?** Not Applicable.
-* **Is the rollout accompanied by any deprecations and/or removals of features,  APIs, fields of API types, flags, etc.?** No.
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Kubelet may fail to start. The kubelet may crash.
+
+###### What specific metrics should inform a rollback?
+
+We can use `cpu_manager_pinning_errors_total` to see all the allocation errors, irrespective of the specific reason though.
+In addition, we can use the logs: the number of pod ending up in Failed for SMTAlignmentError could be used to decide a rollback.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not Applicable.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
 
 ### Monitoring requirements
-* **How can an operator determine if the feature is in use by workloads?**
-  - Inspect the kubelet configuration of the nodes: check feature gates and usage of the new options
-* **What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?**
-  - No change
-* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?** N/A.
-* **Are there any missing metrics that would be useful to have to improve observability if this feature?** N/A.
 
+###### How can an operator determine if the feature is in use by workloads?
+
+- Check the metric `container_aligned_compute_resources_count` with the label `boundary=physical_cpu`
+- Inspect the kubelet configuration of the nodes: check feature gates and usage of the new options
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details:
+    - check metrics and their interplay:
+        * the metric `container_aligned_compute_resources_count` with the label `boundary=physical_cpu`
+        * the metric `cpu_manager_pinning_requests_total`
+        * the metric `cpu_manager_pinning_errors_total`
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
```

Comment:
I don't think these are SLIs.

https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md

Reply:
Are the "errors" here system or user error? If they are system errors, then the error rate (errors/requests) is an SLI. If they are user errors, it's not really. Latency for configuring this could be another SLI, but it is encompassed by the general Pod provisioning SLI and I don't think we need to track this separately.

Since there are no SLOs the SLIs don't mean much. But if this is a *system* error, we probably should use it to specify an SLO as well.

I am OK leaving these in here for documentation, regardless (or they could move to the troubleshooting section)

Reply:
`cpu_manager_pinning_errors_total` are system errors. Meaning: the system was asked to pin CPUs because the workloads requests them, but failed to do so. `cpu_manager_pinning_requests_total` tracks the number of workloads (containers) managed by the kubelet which requested cpu pinning (the usual cpumanager requirement: guaranteed QoS pods, integral CPU requested).

AFAICT I agree there is no SLOs, but still these metrics may be helpful to troubleshoot nodes.

TL;DR: AFAIU I can leave this part unchanged, so I'm doing that. Please let me know if I need to change something after the clarification above.